engine/src/test/java/org/camunda/bpm/engine/test/api/task/TaskServiceTest.java

```diff
@@ -2228,14 +2226,47 @@ public void testDeleteTaskAttachmentWithNullParameters() {
   }
 
   @Test
-  public void testDeleteTaskAttachmentWithTaskIdNull() {
-    int historyLevel = processEngineConfiguration.getHistoryLevel().getId();
-    if (historyLevel> ProcessEngineConfigurationImpl.HISTORYLEVEL_NONE) {
-      try {
-        taskService.deleteTaskAttachment(null, "myAttachmentId");
-        fail("expected process engine exception");
-      } catch(ProcessEngineException e) {}
-    }
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_NONE)
+  public void testDeleteTaskAttachmentThatDoesNotExist() {
+    assertThatThrownBy(() -> taskService.deleteTaskAttachment(null, "attachmentDoesNotExist")).isInstanceOf(
+            NullValueException.class)
+        .hasMessageContaining("No attachment exists with attachmentId 'attachmentDoesNotExist'");
+  }
+
+  @Test
+  @Deployment(resources = { "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml" })
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_NONE)
+  public void testDeleteTaskAttachmentWithTaskIdEmpty() {
+    // given
+    runtimeService.startProcessInstanceByKey("oneTaskProcess");
+    Attachment attachment = taskService.createAttachment("web page", "", null, "weatherforcast",
+        "temperatures and more", new ByteArrayInputStream("someContent".getBytes()));
```

Comment:
Based on my understanding of the issue, I think it would also make sense to look into prohibiting the creation of task attachments with empty taskId and null (or empty) processInstanceId. At the moment, this is still allowed because the creation only [checks](https://github.com/camunda/camunda-bpm-platform/blob/119cbf8804b79fd53310ed42ebd567f6e26812fa/engine/src/main/java/org/camunda/bpm/engine/impl/cmd/CreateAttachmentCmd.java#L71) for the taskId not being null. I can open a follow up ticket you agree that this makes sense

Reply:
Correct, but its more about creating an attachment without taskId and without processInstanceId. My understanding is that it should have either one or the other, but that its fine for it to not have a taskId as long as it still has a processInstanceId.

Reply:
From what we can see in the [create attachment documentation](https://github.com/camunda/camunda-bpm-platform/blob/824b35f3f281d3e8136b2fc204684d3c83ef9bc1/engine/src/main/java/org/camunda/bpm/engine/TaskService.java#L1216), what you say makes sense.

> `@param processInstanceId - id of a process to use if task id is null`

I guess we can create a follow-up ticket to prevent that both `taskId` and `processInstanceId` are null on creation.

Reply:
[Here](https://github.com/camunda/camunda-bpm-platform/issues/4834) is the follow up ticket